### PR TITLE
Fix emulator flags

### LINKED ISSUE #82: beforeEach is undefined
### Problem

(This might be a problem with `flow-cli` rather than `flow-js-testing`)
After running `brew update && brew upgrade flow-cli` the following error appears:

```
thrown: undefined

      52 |
      53 | describe("Project", () => {
    > 54 |   beforeEach(async () => {
         |   ^
      55 |     const basePath = path.resolve(__dirname, "./cadence");
      56 |     const port = 8080;
      57 |     init(basePath, port);

      at Project.test.js:54:3
      at Object.<anonymous> (Project.test.js:53:1)
```

### Steps to Reproduce

Use the latest version of `flow-cli`, and the problem occurs from `flow-js-testing` 0.1.15 to 0.2.2

### Context

Currently working on testing a contract in preparation for deployment to mainnet. The project is ready for release, so this is preventing the last step we need to deploy and holding up the project.

### LINKED ISSUE #83: Error: unknown flag: --http-port
## Description
CLI release `0.31.0` replaced `--http-port` with `--admin-port` flag, which broke emulator process wrapper.

## Solution
Update used flag in emulator wrapper and release new version.

### LINKED ISSUE #85: ShallRevert does not work as expected
### Problem

ShallRevert does not work properly , it won't throw any error whether or not ix execute successfully.

### Description

[https://github.com/onflow/flow-js-testing/blob/master/src/jest-asserts.js#L76](https://github.com/onflow/flow-js-testing/blob/master/src/jest-asserts.js#L76)
If ix succeed , it will return with result, and ``await expect(result).toBe(null)`` shall throw, then the error will be catch, then ``resolvedError = "ERROR!"`` , finally ``await expect(resolvedError).not.toBe(null)`` just pass without any error !

### PR DESCRIPTION
**Pull request by MaxStalker**
Closes #82 
Closes #83
Closed #85 

- Update emulator flags passed to process.
- Enable JSON logging
- Update message and log processing
- Update Jest asserts

### CODE REVIEW

File: test/emulator.test.js [hunk 1]
```diff
@@ -5,14 +5,21 @@ describe("emulator - logging", () => {
...
     const output = emulator.parseDataBuffer(msg);
-    console.log({ output });
+    expect(output.msg).toBe(msg)
+    expect(output.level).toBe("parser")
   });
 
   it("shall format logged message", ()=>{
-    const msg =`time="2021-09-28T15:06:56+03:00" level=info msg="🌱  Starting gRPC server on port 3569" port=3569`
-
+    // const msg =`time="2021-09-28T15:06:56+03:00" level=info msg="🌱  Starting gRPC server on port 3569" port=3569`
```
Comment (kerrywei) [inline test/emulator.test.js:13]: is it intentional to keep this line commented out?

File: src/emulator.js [hunk 2]
```diff
@@ -78,59 +88,69 @@ export class Emulator {
...
+
+        this.log(`EMULATOR ERROR: ${message}`, "error");
         this.initialized = false;
         clearInterval(internalId);
         reject();
       });
 
       this.process.on("close", (code) => {
-        this.log(`emulator exited with code ${code}`);
+        if (this.filters.includes("service")) {
```
Comment (kerrywei) [inline src/emulator.js:151]: is this filtering too broad?

Comment (MaxStalker) [reply src/emulator.js:151]: There is 0 messages emitted with this level by emulator natively.
I was just thinking about the way to filter messages like `process exited with code 2`

File: test/emulator.test.js (hunk 1 above)
Comment (MaxStalker) [reply test/emulator.test.js]: Nope. Forgot to delete, when creating test object.

**Review by kerrywei — APPROVED** [review]

**Pull request merged**